msrestazure/azure_active_directory.py

```diff
@@ -632,10 +644,18 @@ def get_msi_token_webapp(resource):
     return token_entry['token_type'], token_entry['access_token'], token_entry
 
 
+def _is_app_service():
+    # Might be discussed if we think it's not robust enough
+    return 'APPSETTING_WEBSITE_SITE_NAME' in os.environ
```

Comment:
I am not familiar with this env variable, but I know we have `MSI_ENDPOINT` and `MSI_SECRET` to leverage

Reply:
When I did the MSI sample of app service, I realized that I need first to detect if I'm on a WebApp. For that, MSI_ENDPOINT is not good (if MSI is not activated).
APPSETTING_WEBSITE_SITE_NAME is always there, it's a WebApp requirement. This makes possible to detect: if it's a WebApp, no MSI_ENDPOINT is not there, tell the customer.

Reply:
Got it. I don't know whether `APPSETTING_WEBSITE_SITE_NAME ` is reliable moving forward for both Linux and Windows. //CC: @panchagnula 
My opinion was it is the client applications' responsibility to ensure the webapp was enabled with MSI before execute the code, so our library can make such assumption on the existence of env varaibles

Reply:
@lmazuel, verify in both linux and windows webapp,  `APPSETTING_WEBSITE_SITE_NAME` does exist in both